libmamba/tests/src/util/test_iterator.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <doctest/doctest.h>
+
+#include "mamba/util/iterator.hpp"
+
+using namespace mamba::util;
+
+namespace
+{
+    bool greater_than_10(int i)
+    {
+        return i > 10;
+    }
+
+    bool second_greater_than_10(const std::pair<std::string, int>& p)
+    {
+        return p.second > 10;
+    }
+
+    auto get_test_data()
+    {
+        std::vector<int> input_sequence = { 1, 12, 2, 3, 14, 4, 18, 20, 4 };
+        std::vector<int> result_sequence = { 12, 14, 18, 20 };
+        std::map<std::string, int> input_map =
+        {
+            {"a", 1},
+            {"b", 12},
+            {"c", 2},
+            {"f", 3},
+            {"x", 14},
+            {"g", 4},
+            {"d", 18},
+            {"l", 20},
+            {"n", 4}
+        };
+        std::map<std::string, int> result_map =
+        {
+            {"b", 12},
+            {"x", 14},
+            {"d", 18},
+            {"l", 20}
+        };
+
+        return std::make_tuple(input_sequence, result_sequence, input_map, result_map);
+
+    }
+}
+
+TEST_SUITE("filter_iterator")
+{
+    TEST_CASE("forward_iterator_api")
+    {
+        auto [input_seq, res_seq, input_map, res_map] = get_test_data();
```

Comment:
`const` ?

Reply:
I want to test non const iterators, for that I need non const sequences.